### PR TITLE
fixes #430 - Better error messages when missing drivers

### LINKED ISSUE #430: Better error messages when missing drivers
Probably catch ClassNotFoundExceptions and return/throw a nicer error message, that states
which jars should be put into the plugins directory.

- [ ] JDBC
- [ ] MongoDB
- [ ] Couchbase

Also add tot he docs where to download the right drivers, similar to:
https://neo4j-contrib.github.io/neo4j-apoc-procedures/#_load_jdbc_resources

### PR DESCRIPTION
**Pull request by AngeloBusato**
@jexp for jdbc we have already an error who specifies that driver is missing.
If you want, we can create a map that specifies the drive to be inserted into the plugin folder (Oracle, Derby...). 

missingdriverjdbcerror

### CODE REVIEW

**Comment by jexp** [comment]
Perhaps we can still improve it, ala "Please download and copy the JDBC driver into $NEO4J_HOME/plugins,more details at https://neo4j-contrib.github.io/neo4j-apoc-procedures/#_load_jdbc_resources"

**Comment by AngeloBusato** [comment]
@jexp Done

**Comment by jexp** [comment]
I don't like to add a checked exception. If we can turn the "PluginException" into a "MissingDependencyException" which is a RuntimeException, we can just throw it and it shows up for the user. No need to pollute signatures with a checked exception.

**Comment by AngeloBusato** [comment]
@jexp fixed

File: src/main/java/apoc/mongodb/MongoDB.java [hunk 1]
```diff
@@ -62,38 +66,62 @@ private Coll getColl(@Name("host") String hostOrKey, @Name("db") String db, @Nam
...
 
+    private Coll getMongoColl(String hostOrKey, String db, String collection){
+        Coll coll = null;
+        try {
+            coll = getColl(hostOrKey, db, collection);
+        }
+        catch (NoClassDefFoundError e) {
+            throw new MissingDependencyException("Cannot find the jar into the plugins folder. \n"+
+                    "Please put these jar in the plugins folder :\n\n" +
+                    "bson-3.4.2.jar\n" +
```
Comment (jexp) [inline src/main/java/apoc/mongodb/MongoDB.java]: probably leave off the exact versions? and just do -.x.y.z and link to the docs.

File: src/main/java/apoc/couchbase/Couchbase.java [hunk 2]
```diff
@@ -438,4 +440,18 @@
...
     return result;
   }
+
+  private CouchbaseConnection getCouchbaseConnection(List<String> nodes, String bucket){
+    try (CouchbaseConnection couchbaseConnection = CouchbaseManager.getConnection(nodes, bucket)) {
+      return couchbaseConnection;
+    } catch (NoClassDefFoundError e){
+      throw new MissingDependencyException("Cannot find the jar into the plugins folder. \n"+
+              "Please put these jar in the plugins folder : \n\n" +
+              "java-client-2.3.1.jar\n" +
```
Comment (jexp) [inline src/main/java/apoc/couchbase/Couchbase.java]: probably leave off the exact versions? and just do -.x.y.z and link to the docs.

Comment (AngeloBusato) [inline src/main/java/apoc/couchbase/Couchbase.java]: @jexp fixed

File: src/main/java/apoc/mongodb/MongoDB.java (hunk 1 above)
Comment (AngeloBusato) [inline src/main/java/apoc/mongodb/MongoDB.java]: @jexp fixed

**Comment by AngeloBusato** [comment]
@jexp fixed

**Pull request merged**